### PR TITLE
[ZEPPELIN-5174]. NPE in SparkSqlInterpreter

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

The exception in SparkSqlInterpreter may not have cause, so this PR is to fix the NPE via checking whether the cause is null.

### What type of PR is it?
[Bug Fix]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-5174

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by zjffdu** [comment]
Will merge if no more comment

File: spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java [hunk 1]
```diff
@@ -115,14 +115,23 @@ public InterpreterResult internalInterpret(String st, InterpreterContext context
           context.out.flush();
           return new InterpreterResult(Code.ERROR);
         } else {
+          LOGGER.error("Error happens in sql: " + curSql, e);
```
Comment (Reamer) [inline spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java:118]: Please use the parameterized method.

**Review by Reamer — CHANGES_REQUESTED** [review]
Only one small thing.
Everything else LGTM.

File: spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java (hunk 1 above)
Comment (zjffdu) [reply spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java:118]: The last parameter is Exception, I think we can not use parameterized method here.

Comment (Reamer) [reply spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java:118]: You are right.

**Review by Reamer — APPROVED** [review]

File: spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java (hunk 1 above)
Comment (Reamer) [reply spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java:118]: Hi @zjffdu,
I was a little bit confused. It is possible to use parameterised methods for exceptions as well.
Checkout: http://www.slf4j.org/faq.html#paramException

Comment (zjffdu) [reply spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java]: You are right, it is supported, I verified it locally, learned new things :)

Comment (Reamer) [reply spark/interpreter/src/main/java/org/apache/zeppelin/spark/SparkSqlInterpreter.java]: I also learned a lot from the Zeppelin project.

**Comment by zjffdu** [comment]
Will merge if no more comment

**Pull request closed without merging**